src/components/PopupWithForm.js

```diff
@@ -18,27 +19,49 @@ export default class PopUpWithForm extends PopUp {
   }
 
   _handleSubmit(evt) {
-    evt.preventDefault(); 
+    evt.preventDefault();
     const inputValue = this._getInputValues();
-    this._handleFormSubmit(inputValue);
-    this.close();
+    this.setLoading(true);
+    this._handleFormSubmit(inputValue)
+    .then(() => {
+      this.close();
+      this.setLoading(false);
```

Comment:
`      this.setLoading(false);
`


You need to return the **default** button text only in blocks `finally` not to duplicate it 2 times


```js
    this._handleFormSubmit(inputValue)
      .then(() => {
        this.close();
      })
      .catch((err) => {
        console.error(err);
      })
      .finally(() => this.setLoading(false));

```